Library/Homebrew/utils/deprecate.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+module Utils
+  module Deprecate
+    # This module is common for both formulae and casks.
+    # To-do : Add support for casks + add disable method
+
+    def deprecate
+      self.is_deprecated = true
+
+      # deprecate all formulae dependent on this deprecated formula
+      all_formulae = ObjectSpace.each_object(Class).select { |klass| klass < Formula }
```

Comment:
This is the part that has been quite confusing for me. I am basing this off the fact that all formulae are not instances/objects of the `Formula` class, but actually derived classes inherited from the `Formula` class. Am I right here?

Reply:
> Am I right here?

Yup!